ext/mixed/js/core.js

```diff
@@ -288,6 +293,45 @@ const yomichan = (() => {
             this.trigger('orphaned', {error});
         }
 
+        getTemporaryListenerResult(eventHandler, userCallback, timeout=null) {
+            let resolve;
+            let reject;
+            const listenerPromise = new Promise((_resolve, _reject) => {
+                resolve = _resolve;
+                reject = _reject;
+            });
+
+            if (eventHandler === chrome.runtime.onMessage) {
+                const runtimeMessageCallback = ({action, params}, sender, sendResponse) => {
+                    let timeoutId = null;
+                    if (timeout !== null) {
+                        timeoutId = window.setTimeout(() => {
+                            timeoutId = null;
+                            reject(new Error(`Listener timed out in ${timeout} ms`));
+                            eventHandler.removeListener(runtimeMessageCallback);
+                        }, timeout);
+                    }
+
+                    const cleanupResolve = (value) => {
+                        if (timeoutId !== null) {
+                            window.clearTimeout(timeoutId);
```

Comment:
Set timeoutId to null.

Reply:
I had a comment about this earlier:

> Btw, I wonder if clearTimeout actually needs to check if the timer has been cleared. From what I can gather it's idempotent and you can even call it on ID's that never existed without warnings. Performance wise an if (bool) should be faster than a function call (when it's false), but it adds a few lines and some indentation.

In general I agree that it's good to avoid calling functions with unknown side effects or costly processing multiple times, and I can make it null the ID here to be consistent.

Reply:
https://github.com/FooSoft/yomichan/commit/8c32ce7799fac5dc0694ab51ab65d31c3ea76fe1

Reply:
Yeah, it technically doesn't need to be checked or null'd since the IDs are never reused. It was mainly a consistency remark.